src/Internal/UriValidation/UriValidation.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+
+namespace AppInstaller::UriValidation
+{
+    // The decision made based on the Uri validation.
+    enum class UriValidationDecision
+    {
+        Allow,

```

Comment:
Isn't there also a `Warn` response that SmartScreen can return? 

https://learn.microsoft.com/en-us/defender-endpoint/web-protection-overview#order-of-precedence

Reply:
You're right, there's a PUA response as well. However, PUA/Warn will also be treated as Blocked, but the details will be communicated in the logs instead and the logic will be in the injected code. I can potentially reduce this enum to a bool. I will revisit this later once things are a little bit more finalized.